app/models/content_store_payload_version.rb

```diff
@@ -0,0 +1,41 @@
+class ContentStorePayloadVersion < ActiveRecord::Base
+  def self.increment(content_item_id:)
+    sql = <<-SQL
+        INSERT INTO content_store_payload_versions (content_item_id)
+        SELECT #{ content_item_id || "NULL" }
+        WHERE NOT EXISTS(
+        SELECT *
+        FROM content_store_payload_versions
+        WHERE content_item_id #{ content_item_id_sql(content_item_id) });
+
+        UPDATE content_store_payload_versions
+        SET current = COALESCE(current, 0) + 1
+        WHERE content_item_id #{ content_item_id_sql(content_item_id) }
+        RETURNING current;
+      SQL
+
+    ContentStorePayloadVersion.connection
+      .execute(sql)
```

Comment:
Yeah.. I wrestled with this a bit. It's doing the creation in the same call that's the issue. I don't think wrapping it in a transaction in the sql helps with that by my understanding. The read bit 'not existing' could still happen more than once before the write takes place I think. I think it's the `content_item_id` that we need to enforce uniqueness on though. I don't think `current` needs to be involved.